quickjs.c

```diff
@@ -33565,20 +33565,52 @@ JSValue JS_EvalThis(JSContext *ctx, JSValue this_obj,
                     const char *input, size_t input_len,
                     const char *filename, int eval_flags)
 {
+    JSEvalOptions options = {
+        .version = JS_EVAL_OPTIONS_VERSION,
+        .filename = filename,
+        .line_num = 1,
+        .eval_flags = eval_flags
+    };
+    return JS_EvalThis2(ctx, this_obj, input, input_len, &options);
+}
+
+JSValue JS_EvalThis2(JSContext *ctx, JSValue this_obj,
+                     const char *input, size_t input_len,
+                     JSEvalOptions *options)
+{
+    const char *filename = "<unnamed>";
+    int line = 1;
+    int eval_flags = 0;
+    if (options) {
+        if (options->filename)
```

Comment:
Check first if `options->version != JS_EVAL_OPTIONS_VERSION` and throw an exception and return if they don't match.

Why? Imagine a v2 feature is introduced that you use in your project but you're being dynamically linked to an older version of libquickjs that doesn't support it. You almost certainly don't want quickjs to silently ignore it.
```suggestion
        if (options->version != JS_EVAL_OPTIONS_VERSION)
            return JS_ThrowInternalError(ctx, "bad JSEvalOptions version");
        if (options->filename)
```